# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/ipc/tests/test_window_open_discarded_bc.html
# Commit: c49115748677
# Full Hash: c491157486775555a61d748ccbbc054988aacae9
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-03-23 21:38:56
# Description:
#   Bug 1614259 - Ensure BrowisngContexts are detached when nsDocShell is destroyed, r=farre
#   
#   After bug 1582832, DocShell destruction and BrowsingContext detaching happen
#   in separate operations, leaving a gap where a DocShell has been destroyed, but
#   its BrowsingContext is still considered attached. During this gap, the usual
# ==============================================================================

diff -r e8f6b2875121 -r c49115748677 dom/ipc/tests/test_window_open_discarded_bc.html
--- a/dom/ipc/tests/test_window_open_discarded_bc.html	Mon Mar 23 15:19:14 2020 +0000
+++ b/dom/ipc/tests/test_window_open_discarded_bc.html	Mon Mar 23 13:28:09 2020 +0000
@@ -22,13 +22,7 @@
       // BrowsingContext to actually become discarded after we call it, to
       // make sure that the window provider actually has a discarded BC at the
       // end of its own nested event loop.
-      SpecialPowers.Services.tm.spinEventLoopUntil(() => {
-        try {
-          return !win.opener;
-        } catch (e) {
-          return false;
-        }
-      });
+      SpecialPowers.Services.tm.spinEventLoopUntil(() => !win.opener);
     }
   }
   SpecialPowers.addObserver(observer, TOPIC);